commands/template_store_pull.go

```diff
@@ -0,0 +1,66 @@
+package commands
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	defaultTemplateList = "https://raw.githubusercontent.com/martindekov/store/martindekov/unofficial_templates_json/templates.json"
+)
+
+func init() {
+	templateStoreCmd.PersistentFlags().BoolVar(&overwrite, "overwrite", false, "Overwrite existing templates?")
+	templateStoreCmd.PersistentFlags().BoolVar(&pullDebug, "debug", false, "Enable debug output")
+
+	templateStoreCmd.AddCommand(templateStorePullCmd)
+}
+
+var templateStorePullCmd = &cobra.Command{
+	Use:   `pull [TEMPLATE_NAME]`,
+	Short: `Pull templates from store`,
+	Long:  `Pull templates from store supported by openfaas or openfaas-incubator organizations`,
+	Example: `  faas-cli template store pull ruby-http ...
+  faas-cli template store pull openfaas/go openfaas-incubator/node10-express ...`,
+	RunE: runTemplateStorePull,
+}
+
+func runTemplateStorePull(cmd *cobra.Command, args []string) error {
+	if len(args) == 0 {
+		return fmt.Errorf("\nNeed to specify one of the store templates check aviable ones by running the command:\n\nfaas-cli template store list\n")
+	} else {
+		storeTemplates, templatesErr := getTemplateInfo(DefaultTemplateList)
+		if templatesErr != nil {
+			return fmt.Errorf("error while fetching templates from store: %s", templatesErr)
+		}
+
+		for _, template := range args {
+			found := false
+			for _, storeTemplate := range storeTemplates {
+				sourceName := fmt.Sprintf("%s/%s", storeTemplate.Source, storeTemplate.CliLanguage)
+				if template == storeTemplate.CliLanguage || template == sourceName {
+					if overwrite && pullDebug {
+						faasCmd.SetArgs([]string{"template", "pull", storeTemplate.Repository, "--overwrite", "--debug"})
```

Comment:
What is `--debug`?

Reply:
The `template pull` has this option I guess if there is a problem with the template, I really don't know it was there I though someone would know how to use it so I added it

Reply:
I removed those now, I render them unnecessary in our situation since we pull from secure source(the store)